.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,46 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
```

Comment:
should we have the same version here as the project? it is 1.19

Reply:
Thank you. I saw it yesterday, but did not have a chance to update yet. This is a still a draft, WIP. Please ignore any possible noise in this PR for now :)